packages/data/src/resolvers-cache-middleware.js

```diff
@@ -1,53 +1,41 @@
-/**
- * Internal dependencies
- */
-import coreDataStore from './store';
-
 /** @typedef {import('./registry').WPDataRegistry} WPDataRegistry */
 
 /**
  * Creates a middleware handling resolvers cache invalidation.
  *
- * @param {WPDataRegistry} registry   The registry reference for which to create
- *                                    the middleware.
- * @param {string}         reducerKey The namespace for which to create the
- *                                    middleware.
+ * @param {WPDataRegistry} registry  Registry for which to create the middleware.
+ * @param {string}         storeName Name of the store for which to create the middleware.
  *
  * @return {Function} Middleware function.
  */
 const createResolversCacheMiddleware =
-	( registry, reducerKey ) => () => ( next ) => ( action ) => {
-		const resolvers = registry
-			.select( coreDataStore )
-			.getCachedResolvers( reducerKey );
-		Object.entries( resolvers ).forEach(
-			( [ selectorName, resolversByArgs ] ) => {
-				const resolver =
-					registry.stores?.[ reducerKey ]?.resolvers?.[
-						selectorName
-					];
-				if ( ! resolver || ! resolver.shouldInvalidate ) {
+	( registry, storeName ) => () => ( next ) => ( action ) => {
+		const resolvers = registry.select( storeName ).getCachedResolvers();
+		const resolverEntries = Object.entries( resolvers );
+		resolverEntries.forEach( ( [ selectorName, resolversByArgs ] ) => {
+			const resolver =
+				registry.stores[ storeName ]?.resolvers?.[ selectorName ];
+			if ( ! resolver || ! resolver.shouldInvalidate ) {
+				return;
+			}
+			resolversByArgs.forEach( ( value, args ) => {
+				// resolversByArgs is the map Map([ args ] => boolean) storing the cache resolution status for a given selector.
+				// If the value is "finished" or "error" it means this resolver has finished its resolution which means we need
+				// to invalidate it, if it's true it means it's inflight and the invalidation is not necessary.
+				if (
+					( value.status !== 'finished' &&
```

Comment:
Judging by the failing tests, it looks like those optional chaining instances were necessary.

Reply:
Yes, thanks for the hint! The `value` can indeed be `undefined`, although our code always sets it to a valid object value. It's a bug in `EquivalentKeyMap` where `map.delete` merely sets a map entry value to `undefined`:

https://github.com/aduth/equivalent-key-map/blob/d4115f9fbd5d1bf27de58be8d9c0e50eb5ede452/src/equivalent-key-map.js#L204

and the `map.forEach` iterates also through these deleted values, not only through valid ones.